### ISSUE TITLE
Python code crashes(segmentation fault) and cause not exception

### ISSUE THREAD

**Issue by aliartiza75** [python/core-workflow#240]
Hello everyone,
In a unit test I am doing this:
```
parser = MockParser()
parser.parse_args().conf[0] = "test_config"  
```

In the code given above, I am mocking the [ArgParser](https://docs.python.org/3/library/argparse.html)() module and trying to assign a string to the parser's conf parameter for a unit test. The problem with the above code is that it passes sometime and sometime fails and causes the segmentation fault.

The complete code is given below
```
class MockParser:                                                                                                                                                                        │·························
    def parse_args(self):                                                                                                                                                                │·························
        return MockArgs                                                                                                                                                                  │·························
                                                                                                                                                                                         │·························
                                                                                                                                                                                         │·························
class MockArgs:                                                                                                                                                                          │·························
    zookeeper = ["localhost"]                                                                                                                                                            │·························
    kafka = ['localhost:9092']                                                                                                                                                           │·························
    conf = ['sample_test_conf.yml']                                                                                                                                                      │·························
    dev_type = ['dev_type_1']                                                                                                                                                            │·························
    output_topic = ['Topic']                                                                                                                                                             │·························
    tenant_id = ['orion']                                                                                                                                                                │·························
    mode = ['config']                                                                                                                                                                    │·························
    interval = ['10']                                                                                                                                                                    │·························
    zookeeper = ["localhost:2181"]                                                                                                                                                       │·························
    cassandra = ['localhost:9041']                                                                                                                                                       │·························
    output_topic = ['Topic']                                                                                                                                                             │·························
    tenant_id = ['orion']                                                                                                                                                                │·························
    kafka_metrics = ['localhost:9091']                                                                                                                                                   │·························
    poll_interval = [100]                                                                                                                                                                │·························
    conf = ['conf']                                                                                                                                                                      │·························
    service = ['kafka']                                                                                                                                                                  │·························
    service_addr = ['localhost:7777']                                                                                                                                                    │·························
    non_data_fields = ["orion_non_data_fields.txt"]    

```
```
Python version is given below:

$ python --version
  Python 3.5.2  
```

This code run correctly on the python version 3.5.4

**Comment by terryjreedy** [comment]
On 4/30/2018 5:42 AM, Irtiza Ali wrote:

> In a unit test I am doing this:
> 
> |parser = MockParser() parser.parse_args().conf[0] = "test_config" |
[snip]
> |Python version is given below: $ python --version Python 3.5.2 |
> 
> This code run correctly on the python version 3.5.4

I am not sure I understand your point.  You should use 3.5.4, with all 
the extra bug fixes, instead of 3.5.2.
tjr

**Comment by aliartiza75** [comment]
Thank you for the quick response. My purpose was just to report this issue this might help someone else.
